.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,122 @@
+name: CI Unit tests
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  test-legacy-python-versions:
+    
+    name: "Python 2.6 Unit Tests"
+    runs-on: ubuntu-16.04
+
+    strategy:
+      fail-fast: false
+    env:
+      NOSEOPTS: "--verbose"
+
+    steps:
+
+    - name: Install Python 2.6
+      run: |
+        sudo add-apt-repository ppa:deadsnakes/ppa -y
+        sudo apt-get update -y
+        sudo apt install python2.6 python2.6-dev -y
+        sudo ln -sf /usr/bin/python2.6 /usr/bin/python
+        PATH="/usr/bin/${PATH:+:${PATH}}"
+        echo $PATH
+        echo "Installed python2.6"
+        python -V
+        curl https://bootstrap.pypa.io/2.6/get-pip.py -o get-pip.py
+        sudo env "PATH=$PATH" python get-pip.py
+
+    - uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+    
+    - name: Test with nosetests
+      run: |
+        ./ci/nosetests.sh
+        exit $?
+
+  test-current-python-versions:
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          - python-version: 2.7
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+
+          - python-version: 3.4
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+
+          - python-version: 3.6
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+
+          - python-version: 3.7
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            
+          - python-version: 3.8
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+
+          - python-version: 3.9
+            PYLINTOPTS: "--rcfile=ci/3.6.pylintrc"
+            additional-nose-opts: "--with-coverage --cover-erase --cover-inclusive --cover-branches --cover-package=azurelinuxagent"
+    
+    name: "Python ${{ matrix.python-version }} Unit Tests"
+    runs-on: ubuntu-18.04
+
+    env:
+      PYLINTOPTS: ${{ matrix.PYLINTOPTS }}
+      PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
+      NOSEOPTS: "--verbose --with-timer ${{ matrix.additional-nose-opts }}"
+      PYTHON_VERSION: ${{ matrix.python-version }}
+
+    steps:
+    
+    - name: Checkout WALinuxAgent repo
+      uses: actions/checkout@v2
+
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    
+    - name: Install dependencies
+      id: install-dependencies
+      run: |
+        sudo env "PATH=$PATH" python -m pip install --upgrade pip
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+
+    - name: Run pylint
+      run: |
+        pylint $PYLINTOPTS --jobs=0 $PYLINTFILES
+
+    - name: Test with nosetests
+      if: steps.install-dependencies.outcome == 'success'
```

Comment:
Thanks for adding this! Just confirming, but did you verify this works? Like forcing pylint to fail and ensuring the nosetests are still running?

Reply:
I'll run an edited branch and upload a screenshot of the results. One second!

Reply:
As is, it doesn't look like nosetests will still run if pylint fails.

Editing the conditional to `success() || (failure() && steps.install-dependencies.output == 'success'` results in the following, though;

![image](https://user-images.githubusercontent.com/29387136/101958309-eca19b00-3bb7-11eb-801a-403ded40c1b8.png)
![image](https://user-images.githubusercontent.com/29387136/101958358-0216c500-3bb8-11eb-8623-ddc35acb2b48.png)

Reply:
Awesome! Thanks for checking and fixing :)